### ISSUE TITLE
Allow to findBy/findOneBy directly by given object

### ISSUE THREAD

**Issue by clementtalleu** [clementtalleu/php-redis-om#70]
$repository->findBy(['user' => $user]);

**Comment by mehdibo** [comment]
Just taking a look at your project out of curiosity, why do you have multiple ways of fetching an object?

```
$user = $objectManager->find(User::class, 1);

$user = $objectManager->getRepository(User::class)->find(1);
```

**Comment by mehdibo** [comment]
Also, if you can include more details on the issues you opened I'd be happy to help in my spare time

**Comment by clementtalleu** [comment]
@mehdibo The main idea is to imitate and respect the interface of the Doctrine's ObjectManager  :

https://github.com/webmozart/doctrine-common/blob/master/lib/Doctrine/Common/Persistence/ObjectManager.php

So, ideally, you could replace Doctrine's EntityManager with the one provided here without changing the way the application works.

However, the best practice is still to inject the repository to retrieve its objects.

**Comment by clementtalleu** [comment]
For this issue: currently, to retrieve a resource via its nested object, you need to provide the nested object's ID.

$repository->findOneBy(['user' => $user->id]);

What I want to do is to allow to use it this way

$repository->findOneBy(['user' => $user]);

**Comment by mehdibo** [comment]
Ah I see, I don't know if RedisOM is supposed to replace instead of go along with Doctrine but I understand your point.

I'd be happy to give it a try this weekend if you didn't already start on it

**Comment by clementtalleu** [comment]
No, the plan is not to replace Doctrine, but to be able to use the 2 object mappers interchangeably as needed.

**Comment by mehdibo** [comment]
Got it, let me know if I can start on it

**Comment by mehdibo** [comment]
I am currently playing around with the library and trying to implement what you said earlier, would the appropriate approach be:

 - Check if the related object has an index and use it for the lookup
 - If no index is present use the id

I will try to submit a PR this weekend

**Comment by clementtalleu** [comment]
Yes, I think the easiest way is to check if the search criterion is a property that corresponds to an object and if so, retrieve its identifier/index to replace the search parameters.
